runtime/interpreter/interpreter.go

```diff
@@ -4162,6 +4162,28 @@ func (interpreter *Interpreter) checkReferencedResourceNotDestroyed(value Value,
 	})
 }
 
+func (interpreter *Interpreter) checkReferencedResourceNotMovedOrDestroyed(
+	referencedValue Value,
+	locationRange LocationRange,
+) {
+	resourceKindedValue, ok := referencedValue.(ReferenceTrackedResourceKindedValue)
+	if !ok {
+		return
+	}
+
+	if resourceKindedValue.IsDestroyed() {
+		panic(DestroyedResourceError{
+			LocationRange: locationRange,
+		})
```

Comment:
Maybe add a test case covering this branch, in case it's missing

Reply:
This was an already existing check, so should be already covered in existing tests.

Anyway added another, just be sure: 6bfa8213e775eb1e4bb8bd9bc89eccf918dfd251